src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/IClientNotifierService.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer;
+
+internal interface IClientNotifierService
```

Comment:
What makes this a "client notifier service"? This seems like an awkward name. Shouldn't it just be `ILanguageServer`? I've always been confused by the whole "client notifier" thing, since it seems a poor description of something that sends requests and notifications.

Reply:
Naming is hard? Roslyn call's theirs the "[ClientLanguageServerManager](https://github.com/dotnet/roslyn/blob/main/src/Features/LanguageServer/Protocol/IClientLanguageServerManager.cs)", but it certainly doens't manage anything!

I think `ILanguageServer` is not right, as we are the server, and this is talking from the server to the client, so "ClientNotifier" makes some sense to me, though its not really notifying the client, its just sending an LSP request to it.

Reply:
The fact that all of these parameters and fields are called "languageServer" is probably an indicator that something is off. I don't love the word "notifier" here since there's a "SendRequest" method.

Reply:
I agree that ClientLanguageServerManager is a bad name too. :smile:

Reply:
Thinking a bit more broadly about what the purpose of this interface, it's about server-to-client communication, right? Should that be worked into the name?

Reply:
Thanks for the comments anyway. As you can probably tell this PR was done mostly by rename in the IDE, and then a (bad) search and replace to fill the gaps.

I'm thinking some form of  `IClientConnection` might work. The class literally wraps the RPC connection we have to the client, and `clientConnection.SendRequest` seems pretty obvious to me that you're sending a request to the client.

Reply:
I think that works much better.